stone/backends/obj_c_types.py

```diff
@@ -1120,7 +1120,7 @@ def emit_serializer():
                         if is_user_defined_type(data_type):
                             if is_struct_type(data_type) and \
                                     not data_type.has_enumerated_subtypes():
-                                self.emit('jsonDict = [{} mutableCopy];'.
```

Comment:
The only thing i'd be worried about here is that in the old version, we are relying on the fact that we either get a brand new dictionary or nil. I the new code, we are adding the serialized dictionary to whatever `jsonDict` already contains. Are we 100% certain that every instance where this is used does not have an instance of `jsonDict` with some values already in it?? 

Something safer (if your unsure about my question above) would be to do something like:

```
jsonDict = [{} mutableCopy] ?: [[NSMutableDictionary alloc] init];
```

Reply:
this is the deleted code, I think your saying that I should do 
```
let additionDict = [{} mutableCopy] ?: [[NSMutableDictionary alloc] init]; // to guarantee non-nilness
[jsonDict addEntriesFromDictionary: additionDict]
```

Reply:
if we are adding nil entries like `[jsonDict addEntriesFromDictionary:nil]`, the json will still be non-nil. so we don't need a `additionDict` to be non-nil